File: top/chenray/antilitematica/punish/ViolationRecord.java
package top.chenray.antilitematica.punish;

import java.util.UUID;

/**
 * Data class for a player's violation record.
 */
public final class ViolationRecord {
   private final UUID uuid;
   private String playerName;
   private int count;
   private long firstViolation;
   private long lastViolation;
   private int totalViolations;

   public ViolationRecord(UUID uuid, String playerName, int count, long firstViolation, long lastViolation, int totalViolations) {
      this.uuid = uuid;
      this.playerName = playerName;
      this.count = count;
      this.firstViolation = firstViolation;
      this.lastViolation = lastViolation;
      this.totalViolations = totalViolations;
   }

   public UUID uuid() {
      return this.uuid;
   }

   public String playerName() {
      return this.playerName;
   }

   public void playerName(String playerName) {
      this.playerName = playerName;
   }

   public int count() {
      return this.count;
   }

   public void count(int count) {
      this.count = count;
   }

   public long firstViolation() {
      return this.firstViolation;
   }

   public void firstViolation(long firstViolation) {
      this.firstViolation = firstViolation;
   }

   public long lastViolation() {
      return this.lastViolation;
   }

   public void lastViolation(long lastViolation) {
      this.lastViolation = lastViolation;
   }

   public int totalViolations() {
      return this.totalViolations;
   }

   public void totalViolations(int totalViolations) {
      this.totalViolations = totalViolations;
   }
}

File: src/main/kotlin/icu/epochcraft/antilitematica/punish/ViolationRecord.kt
package icu.epochcraft.antilitematica.punish

import java.util.UUID

/**
 * 玩家违规记录（渐进惩罚计数）。
 *
 * @author 阿清
 */
data class ViolationRecord(
    val uuid: UUID,
    val playerName: String,
    /** 当前窗口内违规次数 */
    var count: Int,
    val firstViolation: Long,
    var lastViolation: Long,
    /** 累计总次数（跨窗口不清零） */
    var totalViolations: Int,
    /** 所属世界（null = 全局） */
    val world: String?,
)
